### PR TITLE
[moji]add typings

### PR DESCRIPTION
**Pull request by yasupeke**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/moji/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    start: number;
+    end: number;
+  }
+
+  interface MojisyuRegexp {
+    regexp: RegExp;
+    list: string[];
+  }
+
+  class Moji {
```
Comment (DanielRosenwasser) [inline types/moji/index.d.ts]: While there is a `class` declared in the library, it isn't exposed on the `moji` object itself. Consider converting this to an `interface` and removing the `constructor`.

File: types/moji/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+declare namespace moji {
+  type StringType = 'ZE' | 'HE' | 'ZS' | 'HS' | 'HG' | 'KK' | 'ZK' | 'HK';
+
+  interface MojisyuRange {
+    start: number;
+    end: number;
+  }
+
+  interface MojisyuRegexp {
```
Comment (DanielRosenwasser) [inline types/moji/index.d.ts]: For consistency with `RegExp` consider renaming this to `MojisyuRegExp

File: types/moji/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// Type definitions for moji 0.5
+// Project: https://github.com/niwaringo/moji
+// Definitions by: Yasunori Ohoka <https://github.com/yasupeke>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+declare namespace moji {
+  type StringType = 'ZE' | 'HE' | 'ZS' | 'HS' | 'HG' | 'KK' | 'ZK' | 'HK';
```
Comment (DanielRosenwasser) [inline types/moji/index.d.ts]: Would `CharacterType` be a more descriptive name?

File: types/moji/moji-tests.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import * as moji from 'moji';
```
Comment (DanielRosenwasser) [inline types/moji/moji-tests.ts]: Since `moji` is a function, it should be imported with

```ts
import moji = require('moji');
```

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has unaddressed reviewer comments and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Pull request closed without merging**